src/ec/suite_b/ecdsa/signing.rs

```diff
@@ -136,6 +136,29 @@ impl EcdsaKeyPair {
         Ok(Self::new(alg, key_pair))
     }
 
+    /// Constructs an ECDSA key pair direclty from the private key bytes.
+    ///
+    /// It is recommended to use `EcdsaKeyPair::from_pkcs8()` instead. When
+    /// that is not practical, it is recommended to use
+    /// `EcdsaKeyPair::from_private_key_and_public_key` instead.
+    ///
+    /// Since the public key is not given, the public key will be computed from
+    /// the private key. It is not possible to detect misuse or corruption of
+    /// the private key since the public key isn't given as input.
+    pub fn from_private_key_unchecked(
+        alg: &'static EcdsaSigningAlgorithm,
+        private_key: &[u8],
+    ) -> Result<Self, error::KeyRejected> {
+        let private_key = ec::Seed::from_bytes(
+            alg.curve,
+            untrusted::Input::from(private_key),
+            cpu::features()
+        ).map_err(|_| error::KeyRejected::invalid_encoding())?;
+        ec::KeyPair::derive(private_key)
+            .map(|key_pair| Self::new(alg, key_pair))
+            .map_err(|_| error::KeyRejected::invalid_encoding())
+    }
```

Comment:
In Ed25519, we have:

```rust
    pub fn from_seed_and_public_key(
        seed: &[u8],
        public_key: &[u8],
    ) -> Result<Self, error::KeyRejected> {
        let pair = Self::from_seed_unchecked(seed)?;

        // This implicitly verifies that `public_key` is the right length.
        // XXX: This rejects ~18 keys when they are partially reduced, though
        // those keys are virtually impossible to find.
        if public_key != pair.public_key.as_ref() {
```

That is, `from_seed_and_public_key` is implemented in terms of `from_seed_unchecked`. It seems to me like we should do something similar in ECDSA. What do you think?

Reply:
That seems reasonable. Let me update the PR.